src/meta/src/barrier/creating_job/status.rs

```diff
@@ -12,22 +12,91 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use std::collections::HashMap;
+use std::collections::hash_map::Entry;
+use std::collections::{HashMap, HashSet};
 use std::mem::take;
 use std::sync::Arc;
 
+use risingwave_common::hash::ActorId;
 use risingwave_common::util::epoch::Epoch;
 use risingwave_pb::hummock::HummockVersionStats;
 use risingwave_pb::stream_plan::barrier_mutation::Mutation;
 use risingwave_pb::stream_plan::StreamActor;
-use risingwave_pb::stream_service::barrier_complete_response::CreateMviewProgress;
+use risingwave_pb::stream_service::barrier_complete_response::{
+    CreateMviewProgress, PbCreateMviewProgress,
+};
+use tracing::warn;
 
 use crate::barrier::command::CommandContext;
 use crate::barrier::info::InflightGraphInfo;
 use crate::barrier::progress::CreateMviewProgressTracker;
 use crate::barrier::{BarrierKind, TracedEpoch};
 use crate::manager::WorkerId;
 
+#[derive(Debug)]
+pub(super) struct CreateMviewLogStoreProgressTracker {
+    /// `actor_id` -> `pending_barrier_count`
+    ongoing_actors: HashMap<ActorId, usize>,
+    finished_actors: HashSet<ActorId>,
+}
+
+impl CreateMviewLogStoreProgressTracker {
+    fn new(actors: impl Iterator<Item = ActorId>, initial_pending_count: usize) -> Self {
+        Self {
+            ongoing_actors: HashMap::from_iter(actors.map(|actor| (actor, initial_pending_count))),
+            finished_actors: HashSet::new(),
+        }
+    }
+
+    pub(super) fn gen_ddl_progress(&self) -> String {
+        let sum = self.ongoing_actors.values().sum::<usize>() as f64;
```

Comment:
What's this sum, is it the total barrier count we are pending to backfill from the logstore?

Reply:
The sum is used to calculate the average number of pending barriers.